src/status_im2/contexts/chat/events.cljs

```diff
@@ -338,6 +339,13 @@
   [{:keys [db]} shared-element-id]
   {:db (assoc db :shared-element-id shared-element-id)})
 
+(rf/defn navigate-to-lightbox
+  {:events [:chat.ui/navigate-to-lightbox]}
+  [{:keys [db]} shared-element-id screen-params]
+  (let [updated-db (assoc db :shared-element-id shared-element-id)]
+    (reagent/next-tick #(rf/dispatch [:navigate-to :lightbox screen-params]))
```

Comment:
> And if we need it to happen immediately .. Why no use dispatch-sync?

@ibrkhalil I do not want the `navigate-to` to happen immediately.

> Should we go with https://github.com/day8/re-frame/issues/480#issuecomment-623240473

I just tried this one and it did not work. It navigated without executing the shared element transition. (same for dispatch-later)

Reply:
So I guess a normal `:dispatch` key in the returned map didn't work too .. Cool cool
Thanks Omar

Reply:
> shared-element-id should dispatch first

could you elaborate what does it mean?

Reply:
@ibrkhalil, I don't know the answer to your question. I've never done this workaround with next-tick inside the event handler. It's dirty, but if it's the only thing @OmarBasem found to work for now.

Something tells me if we used `:db` and `:fx` keywords like re-frame suggests, we would be able to correctly order the effects to happen after the `:db`. Unfortunately we can't, unless we change/improve the `rf/defn` macro.

Just for reference:

> In fact, with v1.1.0 best practice changed to event handlers should only return two effects :db and :fx, in which case :db was always done first and then :fx, and within :fx the ordering is sequential. This new approach is more about making it easier to compose event handlers from many smaller functions, but more specificity around ordering was a consequence.

Reply:
@ilmotta I am curious to know what is the difference between what we are currently doing and what you are suggesting here

Reply:
The way we dispatch effects in re-frame is kind of deprecated. But in order to use `:db` and `:fx` we need to change the `rf/defn` macro. I'm curious too @OmarBasem, because I'm not sure if it would solve your problem. You can totally try this if you want, you'll just need to use `re-frame.core/reg-event-fx` instead of `rf/defn`, just don't expect the code to be approved since we currently have a guideline to always use `rf/defn`.